src/services/MakeCsv.php

```diff
@@ -31,12 +38,54 @@ public function getFileName(): string
         return date('Y-m-d') . '-export.elabftw.csv';
     }
 
+ 
+    /**
+     * Populate the entitiesData and metadataColumns arrays
+     */
+    protected function populateEntitiesAndMetadata()
+    {
+        $mdPositions = array();
+        foreach ($this->idArr as $id) {
+           $this->Entity->setId((int) $id);
+           try {
+               $permissions = $this->Entity->getPermissions();
+           } catch (IllegalActionException $e) {
+               continue;
+           }
+           if ($permissions['read']) {
+               $entityData = $this->Entity->entityData;
+               $entityData['url'] = $this->getUrl();
+               $metadataValues = array();
+               if ($entityData['metadata']) {
+                   $decoded = json_decode($entityData['metadata'], true, 512, JSON_THROW_ON_ERROR);
+                   if ($decoded && isset($decoded['extra_fields']) && is_array($decoded['extra_fields'])) {
```

Comment:
don't need to check if `$decoded` is truthy because you have JSON_THROW_ON_ERROR